netlogo-gui/src/main/app/Tabs.scala

```diff
@@ -235,39 +269,88 @@ class Tabs(val workspace:    GUIWorkspace,
     menu.offerAction(newAction)
   }
 
-  private def stripPath(filename: String): String =
-    filename.substring(filename.lastIndexOf(System.getProperty("file.separator")) + 1, filename.length)
-
-  val printAction = {
-    val action = RichAction("print-current-tab") { _ =>
-      currentTab match {
-        case printable: org.nlogo.swing.Printable =>
-          try org.nlogo.swing.PrinterManager.print(printable, workspace.modelNameForDisplay)
-          catch {
-            case abortEx: java.awt.print.PrinterAbortException => org.nlogo.api.Exceptions.ignore(abortEx)
-          }
-      }
-    }
-
-    action.putValue(Action.NAME,            I18N.gui.get("menu.file.print"))
-    action.putValue(ActionCategoryKey,      UserAction.FileCategory)
-    action.putValue(ActionGroupKey,         "org.nlogo.app.Tabs.Print")
-    action.putValue(Action.ACCELERATOR_KEY, UserAction.KeyBindings.keystroke('P', withMenu = true))
-    action
-  }
-
-  override def processMouseMotionEvent(e: java.awt.event.MouseEvent) {
+  override def processMouseMotionEvent(e: MouseEvent) {
     // do nothing.  mouse moves are for some reason causing doLayout to be called in the tabbed
     // components on windows and linux (but not Mac) in java 6 it never did this before and I don't
     // see any reason why it needs to. It's causing flickering in the info tabs on the affected
     // platforms ev 2/2/09
   }
 
-  def handle(e: ModelSavedEvent) {
-    saveExternalFiles()
-  }
-
   def handle(e: AfterLoadEvent) {
     requestFocus()
   }
+
+  object SaveAction extends ExceptionCatchingAction(I18N.gui.get("menu.file.save"), this)
+  with MenuAction {
+    category    = UserAction.FileCategory
+    group       = UserAction.FileSaveGroup
+    accelerator = UserAction.KeyBindings.keystroke('S', withMenu = true)
+
+    @throws(classOf[UserCancelException])
+    override def action(): Unit = currentTab match {
```

Comment:
I've done that. Just wanted to ask, what is the benefit of doing this over acting on the current tab? (taking into consideration, that swapping the actions will happen much more often than clicking "Save")